apps/molgenis-components/src/components/tables/RefSideModal.vue

```diff
@@ -20,66 +21,74 @@
 </template>
 
 <script lang="ts" setup>
-import { ref, toRefs, watch } from "vue";
-import Client from "../../client/client";
+import { Ref, ref, toRefs, watch } from "vue";
+import { IColumn } from "../../Interfaces/IColumn";
 import { IRefModalData } from "../../Interfaces/IRefModalData";
 import { IRow } from "../../Interfaces/IRow";
+import { default as Client, default as client } from "../../client/client";
 import ButtonAction from "../forms/ButtonAction.vue";
 import MessageError from "../forms/MessageError.vue";
 import Spinner from "../layout/Spinner.vue";
 import { getPrimaryKey } from "../utils";
 import RefTable from "./RefTable.vue";
 import SideModal from "./SideModal.vue";
 
-const props = withDefaults(
+const properties = withDefaults(
   defineProps<{
     schema?: string;
     tableId: string;
-    label: string;
+    column: string;
     showDataOwner?: boolean;
     refSchema?: string;
-    rows?: IRow[];
+    rows: IRow[];
   }>(),
   {
     showDataOwner: false,
     rows: () => [] as IRow[],
   }
 );
-const { label, rows, tableId } = toRefs(props);
+const { column, rows, tableId } = toRefs(properties);
 
 const emit = defineEmits(["onClose"]);
 
+let localTableId = ref(tableId.value);
+let localColumn = ref(column.value);
+let localRows = ref(rows.value);
 let loading = ref(true);
-let queryResults = ref([] as IRefModalData[]);
+let queryResults: Ref<IRefModalData[]> = ref([]);
```

Comment:
Ah, IRefModalData is basically multiple IRow? Do we have an IRow? Because semantically we are talking records from the graphql API right matching the refColumn we just clicked?

Reply:
It's an IRow with meta data added onto it. The typing is what we did last sprint, but it doesn't feel right in this story so I think it needs some work.